### ISSUE TITLE
Top with tags and subqueries goes to panic: runtime error: index out of range

### ISSUE THREAD

**Issue by mrqwer88** [influxdata/influxdb#8400]
Hello.

influxdb 1.2.2-1 from repo 
```
deb https://repos.influxdata.com/debian jessie stable
```

Queries with ok result - 
```
select * from (select sum(value) from memoryused where node='evo0.fv.ee' and time > now() - 10m group by ctid)
name: memoryused
time                ctid   sum
----                ----   ---
1495109710826199474 102729 647528448
1495109710826199474 103240 2436841472
1495109710826199474 103301 4201607168
...

select top(sum,1) from (select sum(value) from memoryused where node='evo0.fv.ee' and time > now() - 10m group by ctid)
name: memoryused
time                top
----                ---
1495109702448985715 41213403136

```

But top with tag goes to crash - 
```
select top(sum,ctid,1) from (select sum(value) from memoryused where node='evo0.fv.ee' and time > now() - 10m group by ctid)
ERR: 
```
In log we saw panic - 
```
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: [I] 2017-05-18T11:56:35Z SELECT top(sum, ctid, 1) FROM (SELECT sum(value) FROM containers.autogen.memoryused WHERE node = 'evo0.fv.ee' AND time > now() - 10m GROUP BY ctid) service=query
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: panic: runtime error: index out of range
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: goroutine 1198 [running]:
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: panic(0xa05e00, 0xc4200120a0)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /usr/local/go/src/runtime/panic.go:500 +0x1a1
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.filterIntegerByUniqueTags(0xc44e924800, 0x39, 0x40, 0xc4551a20e0, 0x1, 0x1, 0xbaf1b8, 0x0, 0x0, 0x1100000000)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/call_iterator.go:1050 +0x2b6
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.NewIntegerTopReduceSliceFunc.func1(0xc44e924800, 0x39, 0x40, 0xc460b27ba8, 0xc460b27c98, 0xc460b27c98)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/call_iterator.go:860 +0x45b
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*IntegerSliceFuncReducer).Emit(0xc44e91d540, 0xc460b27ba8, 0xc42176c740, 0xa)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/functions.gen.go:595 +0x44
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*integerReduceIntegerIterator).reduce(0xc45565af00, 0xc42021bef0, 0x456e30, 0xc42021bee0, 0xc42021bee8, 0x686fb9)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:4131 +0x807
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*integerReduceIntegerIterator).Next(0xc45565af00, 0xbb0408, 0xc44ebd1ef0, 0x0)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:4044 +0xa1
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*integerInterruptIterator).Next(0xc4556597e0, 0xc45f8377a0, 0x868e06, 0x180001)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:3432 +0x52
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*bufIntegerIterator).Next(0xc455659800, 0x0, 0x868e30, 0x180001)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:2750 +0x3c
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: github.com/influxdata/influxdb/influxql.(*integerAuxIterator).stream(0xc455659840)
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:3529 +0x32
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: created by github.com/influxdata/influxdb/influxql.(*integerAuxIterator).Start
May 18 14:56:35 infuxdb-ctstat0 influxd[28695]: /root/go/src/github.com/influxdata/influxdb/influxql/iterator.gen.go:3515 +0x3f
May 18 14:56:35 infuxdb-ctstat0 systemd[1]: influxdb.service: main process exited, code=exited, status=2/INVALIDARGUMENT
```

**Comment by jwilder** [comment]
This may already be fixed in master.  @jsternberg

**Comment by jsternberg** [comment]
This is fixed in master and will be included in 1.3.

**Comment by mrqwer88** [comment]
Thank a lot!
